File: tempIdeas/java/src/Erba.java
public class Erba extends Oggetto {
    String vitalita_ ;

    public Erba(Integer livello, Integer rarita, Double spirito, Double vitalita) {
        super(livello,rarita, spirito);
        vitalita_="Vitalità";
        insertStat(vitalita_, vitalita);
        sanitizeInput();
    }

    //METODI DI GET

    public Double getVitalita() {
        return getValoreStat(vitalita_);
    }

    //METODI DI SET
    public Erba clone() {
        return new Erba(getLivello(), getRarita(), getSpirito(), getVitalita());
    }

    //OPERAZIONI

    public Double ricicla() {
        return calcolaMana() / 2 *getValoreStat(vitalita_) * getRarita();
    }

    public void potenzia(Double mana, String parametro) {

        int divisore = 10;
        double incremento = mana / getLivello();

        incrementStat(vitalita_, incremento * getRarita() / divisore);
        incrementStat(parametro, incremento);

        normalizza();
    }

    public void potenzia(Double mana) {
        int divisore = 10;
        double incremento = mana / getLivello();

        List<String> parametri = getListaStats();
        incremento = incremento / parametri.

    }






}

File: tempIdeas/untitled/erba.cpp
#include "erba.h"


Erba::Erba(int livello, int rarita, double spirito, double vitalita) : Oggetto(livello, rarita, spirito), vitalita_("Vitalità") {
    insertStat(vitalita_, vitalita);
}

double Erba::getVitalita() const {
    return getValoreStat(vitalita_);
}

Erba *Erba::clone() const {
    return new Erba(*this);
}

double Erba::ricicla() const {
    return calcolaMana() / 2 + getValoreStat(vitalita_) * getRarita();

}

void Erba::potenzia(double mana, std::string parametro) {

    mana = sanitizeMana(mana);

    int divisore = 10;
    double incremento = mana / getLivello();

    incrementStat(vitalita_, incremento * getRarita() / divisore); //Vitalità riceve un bonus sicuro oltre alla normale distribuzione

    list<string> statsList = getListaStats();
    if((std::find(statsList.begin(), statsList.end(), parametro) == statsList.end())) {

        incremento = incremento / statsList.size();
        for(auto i = statsList.begin(); i != statsList.end(); i++)
            incrementStat(*i, incremento);
    }
    else {
        incrementStat(parametro, incremento);
    }

    normalizza();
}
